examples/service-clients/azure-client/todo-list/src/view.tsx

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	CollaborativeInput,
+	CollaborativeTextArea,
+	SharedStringHelper,
+} from "@fluid-example/example-utils";
+import { useTree } from "@fluid-experimental/tree-react-api";
+import type { IFluidContainer } from "fluid-framework";
+import type { ISharedString, SharedString } from "fluid-framework/legacy";
+import React, { useEffect, useRef, useState } from "react";
+
+import { createTodoItem, type TodoListContainerSchema } from "./fluid.js";
+import type { TodoList, TodoItem } from "./schema.js";
+
+// eslint-disable-next-line import/no-unassigned-import
+import "./style.css";
+
+/**
+ * {@link TodoListAppView} input props.
+ */
+export interface TodoListAppViewProps {
+	readonly todoList: TodoList;
+	readonly container: IFluidContainer<TodoListContainerSchema>;
+}
+
+/**
+ * To-do list application view component.
+ */
+export const TodoListAppView: React.FC<TodoListAppViewProps> = (
+	props: TodoListAppViewProps,
+) => {
+	const { todoList, container } = props;
+
+	const [titleString, setTitleString] = useState<SharedString | undefined>();
+
+	const newItemTextInputRef = useRef<HTMLInputElement>(null);
+	useTree(todoList);
+
+	const todoListTitleHandle = todoList.title;
+
+	useEffect(() => {
+		todoListTitleHandle
+			.get()
+			.then((title) => {
+				setTitleString(title as ISharedString);
+			})
+			.catch((error) => {
+				console.error("Failed to get to-do list title:", error);
+				throw error;
+			});
+		return () => {};
+	}, [todoListTitleHandle]);
+
+	if (titleString === undefined) {
+		return <div>Loading...</div>;
+	}
+	const handleCreateClick = (ev: React.FormEvent<HTMLFormElement>): void => {
+		ev.preventDefault();
+
+		const input = newItemTextInputRef.current;
+		if (input === null) {
+			throw new Error("New item text field missing");
+		}
+
+		createTodoItem({
+			container,
+			initialTitleText: input.value,
+			completed: false,
+		})
+			.then((todoItem) => {
+				todoList.items.insertAtEnd(todoItem);
+				if (input !== null) {
+					input.value = "";
+				}
+			})
+			.catch((error) => {
+				console.error("Failed to create to-do item:", error);
+				throw error;
+			});
+	};
+
+	// Using the list of TodoItem objects, make a list of TodoItemViews.
+	const todoItemViews = todoList.items.map((todoItem, index) => (
+		<div className="item-wrap" key={todoItem.id}>
+			<TodoItemView todoItem={todoItem} />
+			<button
+				className="action-button"
+				onClick={() => {
+					todoList.items.removeAt(index);
```

Comment:
I never really dug down to determine if "remove at index" semantics are 100% safe when considering any "latency" between the model updating vs. the view re-rendering.  Probably doesn't matter for a small demo, but maybe a ST expert would want to determine if there's any best practice long-term.

Reply:
Agreed